File: Java-Project/src/main/java/com/response/BookedRoomResponse.java
package com.response;

import java.time.LocalDate;

import lombok.AllArgsConstructor;


@AllArgsConstructor
public class BookedRoomResponse {
    private Long bookingId;
    private String guestName;
    private LocalDate checkInDate; 
    private LocalDate checkOutDate; 
    private int numOfAdults;
    private int numOfChildren;
    private String guestEmail;
    private boolean isCheckedIn;
    private boolean isBookingConfirmed;
    private String room; 
    private int TotalNumofGuest;


    public BookedRoomResponse(Long bookingId, LocalDate checkInDate, LocalDate checkOutDate,
            boolean isBookingConfirmed) {

        this.bookingId = bookingId;
        this.checkInDate = checkInDate;
        this.checkOutDate = checkOutDate;
        this.isBookingConfirmed = isBookingConfirmed;

    } 


    

}

File: Node-Project/utils/BookedRoomResponse.js
function BookedRoomResponse(bookedRoom) {
    return {
      bookingId: bookedRoom.bookingId,
      guestName: bookedRoom.guestName,
      checkInDate: bookedRoom.checkInDate,
      checkOutDate: bookedRoom.checkOutDate,
      numOfAdults: bookedRoom.numOfAdults,
      numOfChildren: bookedRoom.numOfChildren,
      guestEmail: bookedRoom.guestEmail,
      isCheckedIn: bookedRoom.isCheckedIn,
      isBookingConfirmed: bookedRoom.isBookingConfirmed,
      room: bookedRoom.room ? bookedRoom.room.roomName : null,
      totalNumOfGuest: bookedRoom.numOfAdults + bookedRoom.numOfChildren,
    };
  }
  
  export default BookedRoomResponse;